Maple2.Server.Game/Commands/CommandRouter.cs

```diff
@@ -36,9 +40,39 @@ public void RegisterCommands() {
         var listBuilder = ImmutableList.CreateBuilder<Command>();
         var dictionaryBuilder = ImmutableDictionary.CreateBuilder<string, Command>();
         foreach (Command command in context.Resolve<IEnumerable<Command>>(new NamedParameter("session", session))) {
-            listBuilder.Add(command);
-            foreach (string alias in command.Aliases) {
-                dictionaryBuilder.Add(alias, command);
+            // Check permissions based on command type
+            bool hasPermission = command switch {
+                AnimateNpcCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.Debug) ?? false,
+                AlertCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.Alert) ?? false,
+                BuffCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.GameMaster) ?? false,
+                CoordCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.Debug) ?? false,
+                DailyResetCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.EventManagement) ?? false,
+                DebugCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.Debug) ?? false,
+                FieldCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.Debug) ?? false,
+                FindCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.Find) ?? false,
+                FreeCamCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.Debug) ?? false,
+                HomeCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.Debug) ?? false,
+                GotoCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.Warp) ?? false,
+                ItemCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.SpawnItem) ?? false,
+                KillCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.GameMaster) ?? false,
+                NpcCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.SpawnNpc) ?? false,
+                PetCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.SpawnNpc) ?? false,
+                PlayerCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.PlayerCommands) ?? false,
+                QuestCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.Quest) ?? false,
+                StringBoardCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.StringBoard) ?? false,
+                TriggerCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.Debug) ?? false,
+                TutorialCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.Debug) ?? false,
+                WarpCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.Warp) ?? false,
+                AdminPermissionCommand => session.Player?.AdminPermissions.HasFlag(AdminPermissions.Admin) ?? false,
```

Comment:
could we move the permission to each command instead of this switch?

Reply:
If we do that, the command will still show up to a player who doesn't have permission (but they won't be able to use it) because we're registering the command to the player. I would think we want this to be entirely hidden when a player checks possible commands